reporting/api/src/main/java/org/jboss/windup/reporting/severity/IssueCategoryRegistry.java

```diff
@@ -0,0 +1,170 @@
+            registry = new IssueCategoryRegistry();
+            context.put(IssueCategoryRegistry.class, registry);
+        }
+        return registry;
+    }
+
+    /**
+     * Create a new {@link IssueCategoryRegistry}.
+     */
+    public IssueCategoryRegistry()
+    {
+        // Add some default values as placeholders. These will generally be further defined by the rulesets themselves.
+        addDefaults();
+    }
+
+    /**
+     * Attaches all registered {@link IssueCategory}s to the graph. This allows them to be more easily used
+     * from rules.
+     */
+    public void attachToGraph(GraphContext graphContext)
+    {
+        for (IssueCategory issueCategory : this.issueCategories.values())
+        {
+            IssueCategoryModel model = graphContext.create(IssueCategoryModel.class);
+            model.setCategoryID(issueCategory.getCategoryID());
+            model.setName(issueCategory.getName());
+            model.setDescription(issueCategory.getDescription());
+            model.setOrigin(issueCategory.getOrigin());
+            model.setPriority(issueCategory.getPriority());
+        }
+    }
+
+    /**
+     * Loads the related graph vertex for the given {@link IssueCategory#getCategoryID()}.
+     */
+    public static IssueCategoryModel loadFromGraph(GraphContext graphContext, String issueCategoryID)
+    {
+        return loadFromGraph(graphContext.getFramed(), issueCategoryID);
+    }
+
+    /**
+     * Loads the related graph vertex for the given {@link IssueCategory#getCategoryID()}.
+     */
+    public static IssueCategoryModel loadFromGraph(FramedGraph<EventGraph<TitanGraph>> framedGraph, String issueCategoryID)
+    {
+        Iterable<Vertex> vertices = framedGraph.query().has(WindupVertexFrame.TYPE_PROP, IssueCategoryModel.TYPE)
+                .has(IssueCategoryModel.CATEGORY_ID, issueCategoryID).vertices();
+
+        IssueCategoryModel result = null;
+        for (Vertex vertex : vertices)
+        {
+            if (result != null)
+                throw new DuplicateIssueCategoryException("Found more than one issue category for this id: " + issueCategoryID);
+
+            result = framedGraph.frame(vertex, IssueCategoryModel.class);
+        }
+        return result;
+    }
+
+    /**
+     * Loads the related graph vertex for the given {@link IssueCategory}.
+     */
+    public static IssueCategoryModel loadFromGraph(GraphContext graphContext, IssueCategory issueCategory)
+    {
+        return loadFromGraph(graphContext.getFramed(), issueCategory.getCategoryID());
+    }
+
+    /**
+     * Adds a {@link IssueCategory} to the registry and throws a {@link DuplicateIssueCategoryException} if the item already exists.
+     */
+    public void addCategory(IssueCategory category) throws DuplicateIssueCategoryException
+    {
+        IssueCategory original = this.issueCategories.get(category.getCategoryID());
+
+        // Just overwrite it if it was a placeholder
+        if (!original.isPlaceholder())
```

Comment:
I think you get NPE here once you start adding categories to your XML registry that aren't in the defaults you're adding as placeholders?